src/content/ui-components/components/color-highlight-popover.mdx

```diff
@@ -79,37 +72,197 @@ export default function MyEditor() {
+
+```tsx
+function MyHighlightButton() {
+  const {
+    isVisible,
+    isActive,
+    canColorHighlight,
+    handleColorHighlight,
+    handleRemoveHighlight,
+    label,
+    shortcutKeys,
+    Icon,
+  } = useColorHighlight({
+    editor: myEditor,
+    highlightColor: 'var(--tt-color-highlight-yellow)',
+    label: 'Yellow Highlight',
+    hideWhenUnavailable: true,
+    onApplied: ({ color, label }) => {
+      console.log(`Highlight applied: ${label} (${color})`)
+    },
+  })
+
+  if (!isVisible) return null
+
+  return (
+    <button
+      onClick={handleColorHighlight}
+      disabled={!canColorHighlight}
+      aria-label={label}
+      aria-pressed={isActive}
+    >
+      <Icon />
+      {label}
+      <kbd>{shortcutKeys}</kbd>
+    </button>
+  )
+}
+```
+
+#### Props
+
+| Name                  | Type               | Default     | Description                             |
+| --------------------- | ------------------ | ----------- | --------------------------------------- |
+| `editor`              | `Editor \| null`   | `undefined` | The Tiptap editor instance              |
+| `highlightColor`      | `string`           | `undefined` | CSS color value for the highlight       |
+| `label`               | `string`           | `undefined` | Accessible label for the highlight      |
+| `hideWhenUnavailable` | `boolean`          | `false`     | Hide when highlighting is not available |
+| `onApplied`           | `HighlightHandler` | `undefined` | Callback for highlight changes          |
+
+#### Return Values
+
+| Name                    | Type            | Description                              |
+| ----------------------- | --------------- | ---------------------------------------- |
+| `isVisible`             | `boolean`       | Whether the button should be rendered    |
+| `isActive`              | `boolean`       | If the highlight is currently active     |
+| `canColorHighlight`     | `boolean`       | If highlighting is currently allowed     |
+| `handleColorHighlight`  | `() => boolean` | Function to apply/toggle the highlight   |
+| `handleRemoveHighlight` | `() => boolean` | Function to remove any highlight         |
+| `label`                 | `string`        | Accessible label text for the button     |
+| `shortcutKeys`          | `string`        | Keyboard shortcut (Cmd/Ctrl + Shift + H) |
+| `Icon`                  | `React.FC`      | Icon component for the highlight button  |
+
+## Utility Functions
+
+### `pickHighlightColorsByValue(values)`
+
+Selects specific highlight colors from the built-in palette by their CSS values.
+
+```tsx
+import {
+  pickHighlightColorsByValue,
+  HIGHLIGHT_COLORS,
+} from '@/registry/tiptap-ui/color-highlight-button'
+
+// Select specific colors
+const essentialColors = pickHighlightColorsByValue([
+  'var(--tt-color-highlight-yellow)',
+  'var(--tt-color-highlight-green)',
+  'var(--tt-color-highlight-red)',
+])
+
+// Get all available colors
+const allColors = HIGHLIGHT_COLORS
```

Comment:
Then, what do I do with this variable? What do I use it for? Do I pass it to the Editor as a configuration option?